_studio/mfx_lib/shared/src/libmfxsw_encode.cpp

```diff
@@ -61,6 +64,128 @@
 #endif
 #endif
 
+
+struct CodecKey {
+    const mfxU32 codecId;
+    const bool   fei;
+
+    CodecKey(mfxU32 codecId, bool fei) : codecId(codecId), fei(fei) {}
+
+    friend bool operator<(CodecKey l, CodecKey r)
```

Comment:
Maybe stupid question, but why this is 'friend' and not a struct member? @mgonchar : is that ok?

Reply:
yes, it will work... but I also didn't get the point of friendship. @Alexandr-Konovalov could you comment?

Reply:
`friend` is to define a non-member function (overloaded operator in our case) in the class.
And this is to keep a class and a related operator together.

If you work group prefers making `operator<` a class member, ok, let's do it this way,
it's matter of taste.